### PR TITLE
fix(plugin): derive id from package name

### PR DESCRIPTION
**Pull request by regulusleow**
## Summary

Export plugin `id` in `src/plugin.ts` to satisfy the V1 `PluginModule` shape required by OpenCode's file/path plugin loading path.

**Note:** The type-safety refactor commits that were originally part of this PR have been cherry-picked and merged separately via #93.

### Problem
When loading `opencode-mem` from a local filesystem path (rather than npm install), the V1 plugin module shape requires an explicit `id`. Without it, loading fails with:

`Path plugin ... must export id`

### Changes
- Export plugin `id` in `src/plugin.ts`, derived from `package.json.name` with a safe fallback
- Keep V1 `PluginModule` default export shape: `{ id, server }`
- Add contract test assertions to verify `id` exists, is non-empty, and matches `package.json.name`

### Validation
`bun test tests/plugin-loader-contract.test.ts`

Original work by @regulusleow.

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 import type { PluginModule } from "@opencode-ai/plugin";
+import pkg from "../package.json";
 const { OpenCodeMemPlugin } = await import("./index.js");
+
+export const id =
+  typeof pkg.name === "string" && pkg.name.trim() ? pkg.name.trim() : "opencode-mem";
```
[the remark this answers is not in the document]
Comment (regulusleow) [reply src/plugin.ts:6]: In our current target runtime this plugin is loaded by OpenCode under Bun, so `import pkg from "../package.json"` is valid in practice and does not break local/npm plugin loading in this environment.\n\nThat said, if we later decide to guarantee Node ESM portability for this entrypoint, I can switch this to an assertion/require-safe pattern in a follow-up.

File: src/services/ai/providers/openai-chat-completion.ts [hunk 2]
```diff
@@ -21,10 +27,46 @@ interface ToolCallResponse {
...
+  );
+}
+
+function isToolCallResponse(data: unknown): data is ToolCallResponse {
+  return (
+    typeof data === "object" &&
+    data !== null &&
+    Array.isArray((data as { choices?: unknown }).choices) &&
+    (data as { choices: unknown[] }).choices.length > 0
+  );
```
[the remark this answers is not in the document]
Comment (regulusleow) [reply src/services/ai/providers/openai-chat-completion.ts]: Addressed in the latest commit:

- Renamed `isToolCallResponse` → `hasNonEmptyChoices` to better reflect its actual scope
- Added guard for `choices[0]` itself being a non-null object (prevents the guard from throwing on `choices: [null]`)
- Added validation that `message.content` is `string | null | undefined` and `tool_calls` is `undefined | Array` — the two fields downstream code actually reads

Intentionally stopped there: validating each `tool_call` element's `id / function.name / arguments` shape would go beyond "check fields we actually access" into broad schema validation, so those are left to the static type.

File: src/services/ai/providers/openai-chat-completion.ts [hunk 3]
```diff
@@ -270,7 +328,11 @@ export class OpenAIChatCompletionProvider extends BaseAIProvider {
         }
 
         this.aiSessionManager.addMessage(assistantMsg);
-        messages.push(choice.message);
+        messages.push({
+          role: "assistant",
+          content: choice.message.content || "",
+          tool_calls: choice.message.tool_calls,
+        });
```
[the remark this answers is not in the document]
Comment (regulusleow) [reply src/services/ai/providers/openai-chat-completion.ts:342]: Addressed in the latest commit. Changed to a split approach:

- Session persistence (`assistantMsg.content`): `?? ""` — required because `AIMessage.content` is typed `string` and the DB column is `TEXT NOT NULL`
- API messages array (`messages.push`): `?? null` — preserves OpenAI's `null` for tool-calling responses on the current request

Known trade-off: `null` degrades to `""` on session round-trip, so resumed sessions send `content: ""` instead of `content: null` for older assistant messages. Given that most OpenAI-compatible providers treat these equivalently for assistant messages with `tool_calls`, this is acceptable within the current `AIMessage.content: string` constraint.

**Comment by regulusleow** [comment]
@NaNomicon The Copilot-flagged items have been addressed and tested locally (18/18 passing):

- `isToolCallResponse` → `hasNonEmptyChoices`: renamed and strengthened with minimal field validation
- `content || ""` → split `?? ""` (session) / `?? null` (API messages): preserves OpenAI's `null` for tool-calling responses

As for the `plugin.ts` id change — happy to drop it if it's not needed for mainline usage. Let me know either way.
